### PR TITLE
Add application and expire time to notify-send

### PR DESCRIPTION
**Pull request by genofire**
It is better then `notify-send` as application name.

### CODE REVIEW

**Comment by genofire** [comment]
Why was this PR closed (without any comment) ?

**Comment by marzocchi** [comment]
hello! I closed because I lost of track of it and it fell out of date with master, feel free to rebase and also add support for `-t`, thanks!

**Comment by genofire** [comment]
hey, let me some time to manage it - you need 6 month to take a look into this PR ....

**Comment by genofire** [comment]
Thanks for waiting - what do you think about this version?

File: xdotool/functions [hunk 1]
```diff
@@ -55,14 +55,24 @@ function zsh-notify() {
 
     zstyle -s ':notify:' "$type"-sound notification_sound
     zstyle -s ':notify:' "$type"-icon icon
+    zstyle -s ':notify:' app-name app_name
+    zstyle -s ':notify:' expire-time e_time
```
Comment (marzocchi) [inline xdotool/functions]: i would call the variabile `expire_time`, without abbreviation

File: xdotool/functions [hunk 2]
```diff
@@ -55,14 +55,24 @@ function zsh-notify() {
...
 
     title=$(notification-title "$type" time_elapsed "$time_elapsed")
 
     if [[ -n "$icon" ]]; then
         icon_option="-i $icon"
     fi
 
-    notify-send ${=icon_option} "$title" "$message"
+    if [[ -n "$e_time" ]]; then
+        e_time_option="-t $e_time"
```
Comment (marzocchi) [inline xdotool/functions]: maybe check if `notify-send` supports `-t 0` as a way of saying "no timeout", then you can just set that as the default value in `notify.plugin.zsh` and avoid the "if" here

**Comment by genofire** [comment]
I improve `app-name` by using the first word of notification message.

File: xdotool/functions [hunk 3]
```diff
@@ -55,14 +55,25 @@ function zsh-notify() {
...
+    zstyle -s ':notify:' expire-time expire_time
 
     title=$(notification-title "$type" time_elapsed "$time_elapsed")
 
     if [[ -n "$icon" ]]; then
         icon_option="-i $icon"
     fi
 
-    notify-send ${=icon_option} "$title" "$message"
+    if [[ -n "$app_name" ]]; then
```
Comment (marzocchi) [inline xdotool/functions]: I think we can go all-in and just make it so there's always an `-a` option for `notify-send`, either the first word of the command or a pre-set one.

```suggestion
    if [[ -z "$app_name" ]]; then
        app_name_option="-a ${message%% *}"
        message="${message#* }"
    else
        app_name_option="-a $app_name"
    fi
```

File: xdotool/functions [hunk 4]
```diff
@@ -124,7 +133,7 @@ function store-window-id() {
     # the terminal used to type the command, as this function is run as a prexec
     # hook.
     if [[ -z "$WINDOWID" || "$always_check" == 'yes' ]]; then
-        zstyle ':notify:*' window-id "$(xdotool getwindowfocus)"
+       # zstyle ':notify:*' window-id "$(xdotool getwindowfocus)"
```
Comment (marzocchi) [inline xdotool/functions]: i think we need this, no?

File: xdotool/functions [hunk 5]
```diff
@@ -55,14 +57,21 @@ function zsh-notify() {
...
+    zstyle -s ':notify:' expire-time expire_time
 
     title=$(notification-title "$type" time_elapsed "$time_elapsed")
 
     if [[ -n "$icon" ]]; then
         icon_option="-i $icon"
     fi
 
-    notify-send ${=icon_option} "$title" "$message"
+    if [[ "$app_name" -eq "AUTO" ]]; then
```
Comment (marzocchi) [inline xdotool/functions]: i don't think we need a special value to have an app_name: if you set the default value as an empty string in `notify.plugin.zsh` (as for example `success-sound`), then here you can check if `app_name` is an empty string, and act accordingly.

File: xdotool/functions (hunk 4 above)
Comment (genofire) [reply xdotool/functions]: Oh sorry - i only need this in wayland enviroments (like gnome, sway, ...).
That should not be here in this PR.

**Comment by marzocchi** [comment]
thanks!

**Comment by genofire** [comment]
you are welcome - i do not know how to solve wayland issue yet ...

**Pull request merged**